File: Benchmarking/src/gen416.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: s0,
    p1: Vector[Vector[s0]]
  )
  case class s5 (
    p0: s1,
    p1: Vector[s0]
  )
  case class s6 (
    p0: s5,
    p1: Vector[s1]
  )
  def f98(v0: s0): s0 = {
    var v7: s0 = v0
    var v19: s0 = v7
    v7 = v0
    v19
  }
  def f95(v0: s0): s0 = {
    var v5: s0 = v0
    val v7: Vector[s0] = Vector(v0, v5, v5, v0)
    val v2: s0 = v7(0)
    val v6: s0 = v7(2)
    val v8: s0 = f98(v2)
    var v15: Vector[s0] = v7
    val v13: Vector[Vector[Double]] = v2.p1
    val v10: s0 = v7(0)
    val v12: s0 = v7(3)
    val v14: Vector[Vector[Double]] = v2.p1
    val v9: Vector[Vector[Double]] = v6.p1
    v5 = v5.copy(p0 = v13)
    v15 = v15.updated(2, v8)
    v5 = v5.copy(p0 = v9)
    var v25: Vector[s0] = v7
    v25 = v25.updated(3, v6)
    val v34: Vector[Vector[Double]] = v10.p1
    var v21: Vector[Vector[Double]] = v14
    val v30: Vector[Vector[Double]] = v10.p1
    val v11: s0 = v15(3)
    v5 = v5.copy(p1 = v30)
    val v26: s0 = f98(v12)
    v5 = v5.copy(p0 = v21)
    v25 = v25.updated(3, v8)
    v5 = v5.copy(p0 = v14)
    val v37: s0 = f98(v26)
    v5 = v8
    val v29: Vector[Vector[Double]] = v37.p0
    val v55: Vector[Vector[Double]] = v11.p1
    v5 = v5.copy(p1 = v29)
    v5 = v5.copy(p1 = v9)
    v5 = v5.copy(p0 = v30)
    val v60: Vector[Vector[Double]] = v6.p0
    val v50: Vector[Vector[Double]] = v2.p1
    var v44: Vector[Vector[Double]] = v21
    v25 = v25.updated(0, v8)
    v5 = v5.copy(p0 = v44)
    v5 = v5.copy(p0 = v34)
    v5 = v5.copy(p1 = v60)
    v5 = v5.copy(p0 = v50)
    val v135: Vector[Double] = v55(1)
    val v61: s0 = v25(2)
    v44 = v44.updated(0, v135)
    v61
  }
  def f59(v0: s0): s0 = {
    val v1: s0 = f98(v0)
    val v3: s0 = f98(v1)
    val v5: Vector[Vector[Double]] = v1.p1
    val v6: Vector[Vector[Double]] = v1.p0
    val v7: s0 = f95(v1)
    val v2: s0 = f95(v3)
    var v13: s0 = v7
    var v10: s0 = v13
    v13 = v2
    v10 = v10.copy(p0 = v6)
    v10 = v10.copy(p1 = v6)
    v13 = v13.copy(p0 = v5)
    val v18: Vector[Vector[Double]] = v7.p0
    v13 = v13.copy(p0 = v6)
    val v47: s0 = f95(v2)
    v10 = v10.copy(p0 = v18)
    val v88: Vector[Vector[Double]] = v47.p1
    val v48: s0 = f98(v10)
    v13 = v13.copy(p0 = v88)
    v48
  }
  @noinline
  def f0(v0: Vector[s6], v1: Double): Double = {
    val v2: s6 = v0(0)
    val v10: Vector[s1] = v2.p1
    val v26: s1 = v10(2)
    val v20: s0 = v26.p0
    val v18: s0 = f59(v20)
    val v79: Vector[Vector[Double]] = v18.p1
    val v146: Vector[Double] = v79(1)
    var v105: Vector[Double] = v146
    v105 = v105.updated(0, v1)
    val v96: Double = v105(0)
    v96
  }
  def benchmark(): Unit = {
    val v0: Vector[s6] = Vector(s6(s5(s1(s0(Vector(Vector(0.0), Vector(1.0), Vector(2.0)), Vector(Vector(3.0), Vector(4.0), Vector(5.0))), Vector(Vector(s0(Vector(Vector(6.0), Vector(7.0), Vector(8.0)), Vector(Vector(9.0), Vector(10.0), Vector(11.0)))), Vector(s0(Vector(Vector(12.0), Vector(13.0), Vector(14.0)), Vector(Vector(15.0), Vector(16.0), Vector(17.0)))), Vector(s0(Vector(Vector(18.0), Vector(19.0), Vector(20.0)), Vector(Vector(21.0), Vector(22.0), Vector(23.0)))))), Vector(s0(Vector(Vector(24.0), Vector(25.0), Vector(26.0)), Vector(Vector(27.0), Vector(28.0), Vector(29.0))))), Vector(s1(s0(Vector(Vector(30.0), Vector(31.0), Vector(32.0)), Vector(Vector(33.0), Vector(34.0), Vector(35.0))), Vector(Vector(s0(Vector(Vector(36.0), Vector(37.0), Vector(38.0)), Vector(Vector(39.0), Vector(40.0), Vector(41.0)))), Vector(s0(Vector(Vector(42.0), Vector(43.0), Vector(44.0)), Vector(Vector(45.0), Vector(46.0), Vector(47.0)))), Vector(s0(Vector(Vector(48.0), Vector(49.0), Vector(50.0)), Vector(Vector(51.0), Vector(52.0), Vector(53.0)))))), s1(s0(Vector(Vector(54.0), Vector(55.0), Vector(56.0)), Vector(Vector(57.0), Vector(58.0), Vector(59.0))), Vector(Vector(s0(Vector(Vector(60.0), Vector(61.0), Vector(62.0)), Vector(Vector(63.0), Vector(64.0), Vector(65.0)))), Vector(s0(Vector(Vector(66.0), Vector(67.0), Vector(68.0)), Vector(Vector(69.0), Vector(70.0), Vector(71.0)))), Vector(s0(Vector(Vector(72.0), Vector(73.0), Vector(74.0)), Vector(Vector(75.0), Vector(76.0), Vector(77.0)))))), s1(s0(Vector(Vector(78.0), Vector(79.0), Vector(80.0)), Vector(Vector(81.0), Vector(82.0), Vector(83.0))), Vector(Vector(s0(Vector(Vector(84.0), Vector(85.0), Vector(86.0)), Vector(Vector(87.0), Vector(88.0), Vector(89.0)))), Vector(s0(Vector(Vector(90.0), Vector(91.0), Vector(92.0)), Vector(Vector(93.0), Vector(94.0), Vector(95.0)))), Vector(s0(Vector(Vector(96.0), Vector(97.0), Vector(98.0)), Vector(Vector(99.0), Vector(100.0), Vector(101.0)))))))))
    val v1: Double = 102.0
    val start = nanoTime()
    var result: Double = 102.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}

File: Benchmarking/src/gen416.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: [[s0]]
  }
  struct s5 {
    var p0: s1
    var p1: [s0]
  }
  struct s6 {
    var p0: s5
    var p1: [s1]
  }
  func f98(_ v0: s0) -> s0 {
    var v7: s0 = v0
    var v19: s0 = v7
    v7 = v0
    return v19
  }
  func f95(_ v0: s0) -> s0 {
    var v5: s0 = v0
    let v7: [s0] = [v0, v5, v5, v0]
    let v2: s0 = v7[0]
    let v6: s0 = v7[2]
    let v8: s0 = f98(v2)
    var v15: [s0] = v7
    let v13: [[Double]] = v2.p1
    let v10: s0 = v7[0]
    let v12: s0 = v7[3]
    let v14: [[Double]] = v2.p1
    let v9: [[Double]] = v6.p1
    v5.p0 = v13
    v15[2] = v8
    v5.p0 = v9
    var v25: [s0] = v7
    v25[3] = v6
    let v34: [[Double]] = v10.p1
    var v21: [[Double]] = v14
    let v30: [[Double]] = v10.p1
    let v11: s0 = v15[3]
    v5.p1 = v30
    let v26: s0 = f98(v12)
    v5.p0 = v21
    v25[3] = v8
    v5.p0 = v14
    let v37: s0 = f98(v26)
    v5 = v8
    let v29: [[Double]] = v37.p0
    let v55: [[Double]] = v11.p1
    v5.p1 = v29
    v5.p1 = v9
    v5.p0 = v30
    let v60: [[Double]] = v6.p0
    let v50: [[Double]] = v2.p1
    var v44: [[Double]] = v21
    v25[0] = v8
    v5.p0 = v44
    v5.p0 = v34
    v5.p1 = v60
    v5.p0 = v50
    let v135: [Double] = v55[1]
    let v61: s0 = v25[2]
    v44[0] = v135
    return v61
  }
  func f59(_ v0: s0) -> s0 {
    let v1: s0 = f98(v0)
    let v3: s0 = f98(v1)
    let v5: [[Double]] = v1.p1
    let v6: [[Double]] = v1.p0
    let v7: s0 = f95(v1)
    let v2: s0 = f95(v3)
    var v13: s0 = v7
    var v10: s0 = v13
    v13 = v2
    v10.p0 = v6
    v10.p1 = v6
    v13.p0 = v5
    let v18: [[Double]] = v7.p0
    v13.p0 = v6
    let v47: s0 = f95(v2)
    v10.p0 = v18
    let v88: [[Double]] = v47.p1
    let v48: s0 = f98(v10)
    v13.p0 = v88
    return v48
  }
  @inline(never)
  func f0(_ v0: [s6], _ v1: Double) -> Double {
    let v2: s6 = v0[0]
    let v10: [s1] = v2.p1
    let v26: s1 = v10[2]
    let v20: s0 = v26.p0
    let v18: s0 = f59(v20)
    let v79: [[Double]] = v18.p1
    let v146: [Double] = v79[1]
    var v105: [Double] = v146
    v105[0] = v1
    let v96: Double = v105[0]
    return v96
  }
  func benchmark() {
    let v0: [s6] = [s6(p0: s5(p0: s1(p0: s0(p0: [[0.0], [1.0], [2.0]], p1: [[3.0], [4.0], [5.0]]), p1: [[s0(p0: [[6.0], [7.0], [8.0]], p1: [[9.0], [10.0], [11.0]])], [s0(p0: [[12.0], [13.0], [14.0]], p1: [[15.0], [16.0], [17.0]])], [s0(p0: [[18.0], [19.0], [20.0]], p1: [[21.0], [22.0], [23.0]])]]), p1: [s0(p0: [[24.0], [25.0], [26.0]], p1: [[27.0], [28.0], [29.0]])]), p1: [s1(p0: s0(p0: [[30.0], [31.0], [32.0]], p1: [[33.0], [34.0], [35.0]]), p1: [[s0(p0: [[36.0], [37.0], [38.0]], p1: [[39.0], [40.0], [41.0]])], [s0(p0: [[42.0], [43.0], [44.0]], p1: [[45.0], [46.0], [47.0]])], [s0(p0: [[48.0], [49.0], [50.0]], p1: [[51.0], [52.0], [53.0]])]]), s1(p0: s0(p0: [[54.0], [55.0], [56.0]], p1: [[57.0], [58.0], [59.0]]), p1: [[s0(p0: [[60.0], [61.0], [62.0]], p1: [[63.0], [64.0], [65.0]])], [s0(p0: [[66.0], [67.0], [68.0]], p1: [[69.0], [70.0], [71.0]])], [s0(p0: [[72.0], [73.0], [74.0]], p1: [[75.0], [76.0], [77.0]])]]), s1(p0: s0(p0: [[78.0], [79.0], [80.0]], p1: [[81.0], [82.0], [83.0]]), p1: [[s0(p0: [[84.0], [85.0], [86.0]], p1: [[87.0], [88.0], [89.0]])], [s0(p0: [[90.0], [91.0], [92.0]], p1: [[93.0], [94.0], [95.0]])], [s0(p0: [[96.0], [97.0], [98.0]], p1: [[99.0], [100.0], [101.0]])]])])]
    let v1: Double = 102.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 102.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
